src/Resources/config/services.yml

```diff
@@ -27,6 +27,18 @@ services:
         tags:
             - { name: kernel.cache_warmer }
 
+    contao.controller.abstract_csv_import:
+        class: Contao\CoreBundle\Controller\AbstractCsvImportController
+        abstract: true
+        arguments:
+            - "@database_connection"
+            - "@contao.framework"
+            - "@security.token_storage"
+
+    contao.controller.backend_csv_import:
```

Comment:
You're right, it needs to be public because the legacy code does not use real dependency injection.